src/app/modules/rup/components/elementos/edadGestacional.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit } from '@angular/core';
+import { RUPComponent } from './../core/rup.component';
+import { RupElement } from '.';
+
+@Component({
+    selector: 'rup-calculo-edad-gestacional',
+    templateUrl: 'edadGestacional.html'
+})
+@RupElement('CalculoEdadGestacionalComponent')
+export class CalculoEdadGestacionalComponent extends RUPComponent implements OnInit {
+
+    termCapitalized = '';
+    fechaUltimaMenstruacion: Date;
+    fechaUPMConceptId = '21840007'; // conceptId fecha de último periodo menstrual
+    registro: any = {};
+    alerta = '';
+
+    ngOnInit(){
+        if (!this.soloValores) {
+            // Observa cuando cambia la propiedad 'percentiloPeso' en otro elemento RUP
+            this.conceptObserverService.observe(this.registro).subscribe((data) => {
+                if (this.registro.valor !== data.valor) {
+                    this.registro.valor = data.valor;
+                    this.emitChange(false);
+                    if (this.elementoRUP) {
+                        this.calculoEdadGestacional();
+                    }
+                }
+            });
+        }
+        this.termCapitalized += this.elementoRUP.conceptos[0].term[0].toUpperCase() + this.elementoRUP.conceptos[0].term.slice(1).toLowerCase();
+    }
+
+    cumpleReglasParams(elementoRUP) {
+        if (this.params && this.params.reglas) {
+            if (this.params.reglas.visualizacion) {
+                if (this.params.reglas.visualizacion.ocultar) {
```

Comment:
```suggestion
        if (this.params?.regalas?.visualizacion?.ocultad?) {
```
Queda muy feo o no se puede hacer??

Reply:
no no, para nada. Mala miaaa